HumanGEMInstaller.m

```diff
@@ -0,0 +1,50 @@
+classdef HumanGEMInstaller
+% HumanGEMInstaller
+%   Support for installing and uninstalling
+%   Run HumanGEMInstaller.install to install (will set up the path in MATLAB)
+%   Run HumanGEMInstaller.uninstall to clear the path from MATLAB
+%   To install, you first need to cd to the HumanGEM root.
+
+    methods (Static)
+        function install
+            sourceDir = fileparts(which(mfilename));
+            paths = HumanGEMInstaller.GetFilteredSubPaths(sourceDir, HumanGEMInstaller.FILE_FILTER);
+            addpath(paths);
+            savepath;
+        end
+        function uninstall
+            sourceDir = fileparts(which(mfilename));
+            paths = HumanGEMInstaller.GetFilteredSubPaths(sourceDir, HumanGEMInstaller.FILE_FILTER);
+            rmpath(paths);
+            savepath;
+        end
+        function path = getHumanGEMMainPath()
+			path = fileparts(which(mfilename));
+			path = strrep(path, '\', '/'); %get rid of backslashes in Windows
+			if ~endsWith(path, '/')
+				path = strcat(path,'/');
+			end
+		end
+
+        function newPaths = GetFilteredSubPaths(path_, filter_)
+            pathSep = pathsep();
+			%Check that there are no separators in the path - that will cause 
+            %problems since the separator is used to separate paths in a string
+			if contains(path_, pathSep)
+				error('The path in which Human-GEM resides may not contain path separator chars such as semicolon for this installation to work!');
+			end
+            paths = genpath(path_);
+            splitPaths = strsplit(paths, pathSep);
+            %remove the last, it is empty
+            splitPaths = splitPaths(1,1:end-1);
+            matches = regexp(splitPaths, filter_, 'match');
+            okPaths = cellfun(@isempty, matches);
+            pathsLeft = splitPaths(1,okPaths);
+            newPaths = strcat(char(join(pathsLeft, pathSep)), pathSep);
+        end
+    end
+    
+    properties (Constant)
+      FILE_FILTER = '.*\.git.*'; %should we add ".deprecated" here?
```

Comment:
A more general filter would be to ignore all files and folder that start with `.`. I would also ignore everything ending in `.md`.

One way is to use a recursive negative filter, like done here. Another would be to specify the folders which may be recursively added.

Reply:
So, I prefer this one in general, because it will just work if you add another folder that should be added (which I'm thinking is the more common case?). I don't think it matters super much which. But changing the filter to "." sounds like a good idea, I will experiment with this.

Reply:
Hmm, maybe you are right, there are many other folders such as data etc. that don't need to be added. But I don't think it is a big thing - this solves the main issue. I did remove the .deprecated - it was a bit complicated to filter on just */.*, since it gets a bit complicated with slashes on Windows and Linux (I cannot match on only ".", then everything containing a dot will be lost), and I had no way of testing both. So, I left it like this. Another future improvement would be to check if there is another Human-GEM installed and warn if there is, that is probably more important. But I'm thinking we could leave it like this? I have many things to attend to these days :)